kernel/src/lapic.c

```diff
@@ -0,0 +1,21 @@
+#include "lapic.h"
+
+#include "madt.h"
+#include "memory/paging.h"
+
+// This approach should work as long as all processors with kernel access use the same page tables.
+
+// Paged pointer to the physical xAPIC MMIO interface.
+volatile XAPIC* g_lapic;
```

Comment:
The struct itself should be marked as volatile instead of the individual variables